serializer/custom_normalizer.rst

```diff
@@ -0,0 +1,106 @@
+.. index::
+   single: Serializer; Custom normalizers
+
+How to Create your Custom Normalizer
+====================================
+
+The :doc:`Serializer Component </components/serializer>` uses Normalizers
+to transform any data to an array.
+
+The Component provides several built-in normalizer that are described
+:doc:`in their own section </serializer/normalizers>` but you may want
+to use another structure that's not supported.
+
+Creating a new normalizer
+-------------------------
+
+Imagine you want add, modify, or remove some properties during the serialization
+process. For that you'll have to create your own normalizer. But it's usually
+preferable to let Symfony normalize the object, then hook into the normalization
+to customize the normalized data. To do that, we leverage the ObjectNormalizer::
+
+    namespace AppBundle\Serializer;
+
+    use AppBundle\Entity\Topic;
+    use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+    use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+    use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+    class TopicNormalizer implements NormalizerInterface
+    {
+        private $router;
+        private $normalizer;
+
+        public function __construct(UrlGeneratorInterface $router, ObjectNormalizer $normalizer)
+        {
+            $this->router = $router;
+            $this->normalizer = $normalizer;
+        }
+
+        public function normalize($topic, $format = null, array $context = array())
+        {
+            $data = $this->normalizer->normalize($topic, $format, $context);
+
+            // Here, add, edit, or delete some data:
+            $data['href']['self'] = $this
+                ->router
+                ->generate(
+                    'topic_show',
+                    ['id' => $topic->getId()],
+                    UrlGeneratorInterface::ABSOLUTE_URL
+                )
+            ;
+
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null)
+        {
+            return $data instanceof Topic;
+        }
+    }
+
+Registering it in your app
```

Comment:
A question for when we merge this into upper branches. Does Symfony autoconfiguration support Normalizers? Thanks!

Reply:
Yes it does :heart: 
I already planned to open a new PR to remove theses for new Symfony versions